### PR TITLE
IJSDK-648A: Update to use PhpStorm

### PR DESCRIPTION
**Pull request by JohnHake**
@pauleveritt - please review these sections in particular:
* Configuring Build.Gradle to Target Products other than IntelliJ IDEA
* Configuring Plugin Projects using a Product-Specific Attribute

### CODE REVIEW

File: products/dev_alternate_products.md [hunk 1]
```diff
@@ -2,65 +2,96 @@
...
+Modifications are needed to the skeleton project's `build.gradle` and `plugin.xml` files, as described below, and on the individual product pages in Part VIII.
+The `build.gradle` file is modified to specify the target product, which determines the APIs available during development.
+The `plugin.xml` file is modified to declare the plugin's dependency on modules or libraries.
+
+## Configuring Build.Gradle to Target Products other than IntelliJ IDEA 
+The best practice is to use the IntelliJ IDEA Gradle plugin `intellij.type` [attribute](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/README.md#intellij-platform-properties) to specify the target product.
+For example, `PY` for PyCharm professional.
+Configuration using an `intellij.type` attribute is explained in the [Product-Specific Attribute](#configuring-plugin-projects-using-a-product-specific-attribute) section below.
+
+However, not all products have an `intellij.type` attribute defined by the IntelliJ IDEA Gradle plugin.
```
Comment (YannCebron) [inline products/dev_alternate_products.md]: the name of the plugin is "gradle-intellij-plugin" (multiple occurrences)

File: products/dev_alternate_products.md [hunk 2]
```diff
@@ -2,65 +2,96 @@
...
+```groovy
+    intellij {
+      version '2019.2.3'
+      type 'PY'
+    }
+```
+
+### Configuring Plugin Projects using the IntelliJ IDEA Product Attribute
+If the IntelliJ IDEA Gradle plugin does not directly support an IntelliJ Platform-based product, the `build.gradle` file can still be configured to target the desired product.
+In this case, the `build.gradle` file is configured to use IntelliJ IDEA as the basis for the available APIs.
```
Comment (YannCebron) [inline products/dev_alternate_products.md]: add "Community Edition"

File: products/dev_alternate_products.md [hunk 3]
```diff
@@ -71,30 +102,30 @@ The best practice is to modify the `runIde {}` task to use a local installation
...
     // Define IntelliJ Platform against which to build the plugin project.
-    // Use the IntelliJ Platform BRANCH.BUILD version ("baseIntelliJPlatformVersion") matching "targetIDE"
-    // For example, '192.6817.14'
-    version '<baseIntelliJPlatformVersion>'     
+    // Use the IntelliJ Platform BRANCH.BUILD version matching "targetIDE" (PhpStorm)
+    version '192.7142.36'   // baseIntelliJPlatformVersion     
     type 'IC'                 
     // Require the targetIDE plugin or library 
-    plugins '<targetIDE-specific modules or plugins>'  // For example, 'python' 
+    plugins 'com.jetbrains.php:192.6603.42'   // Stable version of plugin compatible with IntelliJ IDEA build
```
Comment (YannCebron) [inline products/dev_alternate_products.md]: compatible with ... _specified above_

File: products/dev_alternate_products.md [hunk 4]
```diff
@@ -71,30 +102,30 @@ The best practice is to modify the `runIde {}` task to use a local installation
...
   }
   
   runIde {
       // Absolute path to installed target application to use as IDE Development Instance
-      // For example, the path to PyCharm build 192.6817.19
-      ideaDirectory '/Applications/<targetIDE>.app/Contents'
+      // For example, the path to PhpStorm build 192.7142.41, note macOS dialect.
```
Comment (YannCebron) [inline products/dev_alternate_products.md]: pls explain "macOS dialect"

File: products/dev_alternate_products.md (hunk 1 above)
Comment (JohnHake) [reply products/dev_alternate_products.md]: I used "IntelliJ IDEA Gradle plugin" extensively in tutorials/buildsystem/prerequisites.md and gradle_config.md, but I'm agnostic.

File: products/dev_alternate_products.md (hunk 4 above)
Comment (JohnHake) [reply products/dev_alternate_products.md]:
```
 runIde {
      // Absolute path to the installed targetIDE to use as IDE Development Instance
      // Note the Contents directory must be added at the end of the path for macOS.
      ideaDirectory '/Users/jhake/Library/Application Support/JetBrains/Toolbox/apps/PhpStorm/ch-0/192.7142.41/PhpStorm.app/Contents'
  }
```

File: products/dev_alternate_products.md (hunk 2 above)
Comment (JohnHake) [reply products/dev_alternate_products.md]: To be consistent with line 94, I changed line 53 to:
"In this case, the `build.gradle` file is configured to use IntelliJ IDEA (Community or Ultimate Edition) as the basis for the available APIs."

**Pull request merged**